docs/tactical/DESIGN.md

```diff
@@ -0,0 +1,102 @@
+
+### Tactics
+
+Tactics, after which the library is named, are a means of embedding application specific login into Tactical. This knowledge can be used to:
+Resolve queries not cached explicitly in local store.
+Propagate mutations across object relationships.
+Prefetch additional data to build a better local store.
+
+To assist with this process, Tactical exposes a much larger API for use inside of tactics. Through this API, tactics will have the opportunity to hook into any of the components that Tactical employs and will have access to much more of the local model. This will make it easier for consumers to write tactics that will interface with Tactical while still preventing them from rewriting Tactical in order to achieve their goals.
+
+## Why Tactical?
+
+Two other popular options that can be used to solve similar challenges are that of Firebase and Falcor.
+
+### Firebase
+
+Firebase is a cloud-based backend platform that focuses on solving many of the challenges that arise with having a large user base such as authentication and data synchronization. They also announced at Google I/O that they are currently working towards providing a persistent offline model for their consumers to use.
+
+The two main differences between Firebase and Tactical are the expectations we place on the backend model and our approach to data synchronization. Firebase is a backend platform. Their expectation is that, for any application, they will provide all of the backend support that is required. For new applications, Firebase is a great way to move forward quickly with all of the backend support that most applications would ever need. However, for applications that are already in place, using Firebase often means moving all backend services to their platform. Tactical is designed to be integrated alongside an existing data model. Tactical specifications require the ability to handle and compare JSON objects — something that most modern web backends already support.
+
+Second, Firebase’s approach to data synchronization is that the last received request always wins. This means that if multiple users are trying to change the same object, then the last user to submit their change will overwrite all of the previous changes. Tactical takes the opposite approach. Using versioning, Tactical only accepts changes to the version that it has in store. Therefore, when multiple requests attempt to change the same version of an object, only the first request will be accepted. All further requests will be rejected and then Tactical will inform the application of the rejection so that it can resolve the request in a manner it deems the most appropriate.
```

Comment:
Add to this paragraph (last sentence) something about how usually, that involves a simple merge to "rebase" the mutation on top of the latest version.